web-client/src/presenter/actions/CaseConsolidation/updateCaseCheckboxAction.test.js

```diff
@@ -0,0 +1,113 @@
+import { MOCK_CASE } from '../../../../../shared/src/test/mockCase';
+import { runAction } from 'cerebral/test';
+import { updateCaseCheckboxAction } from './updateCaseCheckboxAction';
+
+describe('updateCaseCheckboxAction', () => {
+  it("should flip the correct case's checked state", async () => {
+    const unchangedCheckValue = true;
+    const changedCheckValue = false;
+
+    const customizedDocketNumberOne = '1337-42';
+    const customizedDocketNumberTwo = '1234-42';
+
+    const result = await runAction(updateCaseCheckboxAction, {
+      props: {
+        docketNumber: MOCK_CASE.docketNumber,
+      },
+      state: {
+        caseDetail: {
+          ...MOCK_CASE,
+          consolidatedCases: [
+            {
+              ...MOCK_CASE,
+              checked: unchangedCheckValue,
+              docketNumber: customizedDocketNumberOne,
+              leadDocketNumber: customizedDocketNumberOne,
+            },
+            {
+              ...MOCK_CASE,
+              checked: changedCheckValue,
+              leadDocketNumber: customizedDocketNumberOne,
+            },
+            {
+              ...MOCK_CASE,
+              checked: unchangedCheckValue,
+              docketNumber: customizedDocketNumberTwo,
+              leadDocketNumber: customizedDocketNumberOne,
+            },
```

Comment:
⛏️❓(nit and a question) I might be missing something, but is this third case giving us any more information into a different code path in the logic? It looks to me like it would give us the same info as the first case.

Reply:
Since the lead case always remains checked, we need two additional cases to test the checkboxes on sub-cases. This third case is the _second_ sub-cases.

Reply:
Ah thanks!